Sources/SwiftSyntaxBuilderGeneration/SyntaxBuildableNode.swift

```diff
@@ -0,0 +1,53 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2022 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+/// Extension to the `Node` type to provide functionality specific to
+/// SwiftSyntaxBuilder.
+extension Node {
```

Comment:
While we could also move these types into `Node` directly, I kept them in an extension for clarity (the same goes for `Child` and its extension). I don't have a strong opinion here though, so if you think it's cleaner to have them all in one place, we could also move them to their respective classes.

Reply:
SGTM. Eventually we might want to extract `Node` etc. to a different target if we want to also generate other parts of SwiftSyntax using `SwiftSyntaxBuilder`. But these extensions are specific to the generation of `SwiftSyntaxBuilder` itself.